### PR TITLE
API spec fixes

### LINKED ISSUE #29: "resource.resourcePath" is no equal to "request.target" in "onretrieve/onupdate/ondelete" method
According to [specification](https://github.com/01org/iot-js-api/blob/master/api/ocf/server.md), describe the [`onretrieve`](https://github.com/01org/iot-js-api/blob/master/api/ocf/server.md#onretrieve) method as below:
```
Find the ServerResource object resource for which resource.resourcePath is equal to request.target
```
The `resource.resourcePath` is a property of dictionary `ResourceId`, however `request.target` return `ResourceId `. It is obvious that they are not equal.

The [`onupdate`](https://github.com/01org/iot-js-api/blob/master/api/ocf/server.md#the-onupdatehandler-method) method and [`ondelete`](https://github.com/01org/iot-js-api/blob/master/api/ocf/server.md#the-ondeletehandler-method) method has same error as `onretrieve` method. The correct content should be `Find the ServerResource object resource for which resource.resourcePath is equal to request.target.resourcePath`. I think maybe they are typos, since the `oncreate` method is right.

### PR DESCRIPTION
**Pull request by zolkis**
Fix #29, OCF server retrieve steps

### CODE REVIEW

**Comment by zolkis** [comment]
@nagineni @poussa  could you please review? We'd need to merge this.

File: api/board/spi.md [hunk 1]
```diff
@@ -66,14 +85,17 @@ This internal algorithm is used by the [`Board.spi()`](./README.md/#spi) method.
...
+- If `options.phase` is 0, then set `spi.phase` to 0, otherwise set `spi.phase` to 1.
 - Request the underlying platform to initialize the SPI `spi.bus` with `spi.speed` on `board`.
-- In case of failure, return `null`.
+- In case of failure in any of the steps above, return `null`.
+- Set `spi.frameGap` to 0. Request the underlying platform to provide the SPI inter-frame delay value expressed in nanoseconds and if the request successfully completes, then set `spi.frameGap` to that value.
+- Set `spi.direction` to `"full-duplex"`. Request the underlying platform to provide the SPI transfer mode and if the request successfully completes, then set `spi.direction` to the corresponding value.
 - Return `spi`.
 
-##### The `transfer(device, buffer)` method
+##### The `transceive(device, buffer)` method
```
Comment (nagineni) [inline api/board/spi.md]: Do we really need to pass the 'device' here?

File: api/board/webidl.md [hunk 2]
```diff
@@ -3,13 +3,14 @@ Web IDL for Board and IO APIs
 ```javascript
 
 interface Board {
+    readonly attribute String name;  // board name, e.g. "arduino101"
     attribute EventHandler onerror;
 
-    Pin pin(PinName);  // board-specific dictionary for mapping pins
-    sequence<PinName> pins();  // get all board pin names
+    // Pin pin(PinName);  // board-specific dictionary for mapping pins
+    // sequence<PinName> pins();  // get all board pin names
```
Comment (nagineni) [inline api/board/webidl.md]: Remove the commented lines.

File: api/board/webidl.md [hunk 3]
```diff
@@ -3,13 +3,14 @@ Web IDL for Board and IO APIs
...
 
     AIO aio(PinName pin);
-    GPIO gpio(GPIOInit init);
+    GPIO gpio(PinName or GPIOOptions options);
     PWM pwm( (PinName or PWMOptions) options);
```
Comment (nagineni) [inline api/board/webidl.md]: Remove extra parentheses here.

File: api/board/README.md [hunk 4]
```diff
@@ -55,10 +55,18 @@ The `address` property is the operating-system-specific representation for that
...
+- `"analog-input"` for analog input (AIO) that is converted to digital value.
+- `"analog-output"` for analog output (AIO) that is converted from digital value.
 - `"pwm"` for PWM analog output.
+- `"uart-rx"` for serial receive pin.
+- `"uart-tx"` for serial transmit pin.
+- `"i2c-scl"` for I2C clock.
+- `"i2c-scl"` for I2C data.
+- `"spi-sclk"` for SPI clock.
+- `"spi-mosi"` for SPI Master Out Slave In.
+- `"spi-miso"` for SPI Master In Slave Out.
```
Comment (zolkis) [inline api/board/README.md]: There are 2 possibilities we expose SPI.
1) Specify and SPI `bus` number/index (SCLK+MISO+MOSI), and then a `device` number/index to address the slave (SS) during a `transceive()` operation. The bus number is board specific, and implementation maps it to the physical pin configuration and the MOSI, MISO, SCLK pins are exposed (we could do without the latter).
2) Specify an array of 3+x `Pin` objects, where x >= 1 and denote the SS pins. This is also board specific, but it's not hidden behind bus number and device number.

File: api/board/spi.md (hunk 1 above)
Comment (zolkis) [inline api/board/spi.md]: Depends on which way we expose SPI, see above.

File: api/board/README.md [hunk 5]
```diff
@@ -55,10 +55,18 @@ The `address` property is the operating-system-specific representation for that
...
 <a name="pinmode">
 The `mode` property can take the following values:
-- `"input"` for digital input (GPIO). The pin value can be 0 or 1.
-- `"output"` for digital output (GPIO). The pin value can be 0 or 1.
-- `"analog"` for analog input (AIO) that is converted to digital value.
+- `"digital-input"` for digital input (GPIO). The pin value can be 0 or 1.
+- `"digital-output"` for digital output (GPIO). The pin value can be 0 or 1.
+- `"analog-input"` for analog input (AIO) that is converted to digital value.
+- `"analog-output"` for analog output (AIO) that is converted from digital value.
 - `"pwm"` for PWM analog output.
```
Comment (zolkis) [inline api/board/README.md]: In my understanding PWM works with a digital output (0 or 1) alternated with a given frequency and duty cycle. An analog out pin exposes an analog voltage (rail to rail) based on D/A conversion of a digital sample series/stream.
We could remove the latter for now, but even then the name `analog-input` would stay in order to maintain forward compatibility and correctness of classification.

**Comment by gabrielschulhof** [comment]
OCF LGTM

**Comment by zolkis** [comment]
API versioning will be added in a different PR, as discussed with @gabrielschulhof

**Comment by nagineni** [comment]
@zolkis I was expecting a different PR with API versioning to master before merging this PR and a new version number with the changes in this PR.

**Comment by zolkis** [comment]
There will be bigger changes than just that, and merging would have been hell. 
Still, this API isn't "official" yet, outside of iotivity-node.
Check [my fork](https://github.com/zolkis/iot-js-api) for what's coming up - comments are welcome there, too.

**Pull request merged**